sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -451,8 +451,10 @@ abstract class StreamExecution(
         // Offset has been reached and the equality condition might never be met.
         if (!localCommittedOffsets.contains(source)) {
           true
-        } else if (newOffset.isInstanceOf[LongOffset]) {
-          localCommittedOffsets(source).toString.toLong < newOffset.asInstanceOf[LongOffset].offset
+        } else if (localCommittedOffsets(source).isInstanceOf[LongOffset] &&
```

Comment:
This is going to be verbose on checking the type and casting, so could you please check whether we can enjoy the benefit of pattern matching or not?

```
(localCommittedOffsets.get(source), newOffset) match {
  case (Some(localOff: LongOffset), newOff: LongOffset) => localOff.offset < newOff.offset
  case (Some(localOff), newOff) => localOff != newOff
  case (None, newOff) => true
}
```

Reply:
Verified and it does seem to work in this case. Made the change

Reply:
Simplified this further, based on convo with @HeartSaVioR